src/controller/providers/providers.controller.ts

```diff
@@ -0,0 +1,26 @@
+import { Controller, Get, Logger } from '@nestjs/common';
+import { ApiOkResponse, ApiOperation } from '@nestjs/swagger';
+import { StorageProviderService } from 'src/service/storage-provider/storage-provider.service';
+import { StorageProviderWithIpInfo } from 'src/service/storage-provider/types.storage-provider';
+
+@Controller('providers')
+export class ProvidersController {
+  private readonly logger = new Logger(ProvidersController.name);
+
+  constructor(
+    private readonly storageProviderService: StorageProviderService,
+  ) {}
+
+  @Get('')
+  @ApiOperation({
+    summary: 'Get list of providers with ip info',
```

Comment:
I like to be more eplicit with `storage provider` instead of `provider`, because `provider` seems very generic. So here:
`Get list of storage providers with ip info`

Reply:
Done.